components/org.wso2.carbon.metrics.data.service/src/main/java/org/wso2/carbon/metrics/data/service/MetricsDataService.java

```diff
@@ -191,6 +185,37 @@ private long getStartTime(String from) {
         return sourcesList.toArray(new String[sourcesList.size()]);
     }
 
+    public MetricHierarchyData getHierarchy(String source, String path) {
+        Map<String, MetricType> hierarchicalMetrics = reporterDAO.queryHierarchicalMetrics(source, path);
+        Set<String> childrenNames = new TreeSet<>();
+        List<MetricMeta> metrics = new ArrayList<>();
+        path = (path != null && !path.isEmpty()) ? path : "";
+        for (Map.Entry<String, MetricType> entry : hierarchicalMetrics.entrySet()) {
+            String metric = entry.getKey();
+            MetricType type = entry.getValue();
+            if (metric.startsWith(path)) {
+                String suffix = (!"".equals(path)) ? metric.replaceFirst(path + "\\.", "") : metric;
+                int chunks = suffix.length() - suffix.replace(".", "").length();
+                if (chunks == 0) {
+                    // metrics
```

Comment:
What's this comment supposed to mean? Either make it meaningful or get rid of it altogether.